print-dhcp6.c

```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
 			ND_PRINT(")");
 			break;
 
+		case DH6OPT_BOOTFILE_URL:
+			if (optlen < 2) {
+				ND_PRINT(" ?)");
+				break;
+			}
+			tp = (const u_char *)(dh6o + 1);
+			ND_PRINT(" ");
+			nd_printjnp(ndo, tp, optlen);
```

Comment:
According to RFC 5970 Section 3.1, this option value is not NUL-terminated. If it is not supposed to be zero-padded either, `nd_printjn()` would be a better fit because it would make any out-of-place NUL characters apparent.

Reply:
I think these were copied from other examples in this file for values that are not NUL-terminated, e.g., there's no indication in https://datatracker.ietf.org/doc/html/rfc4833 that the timezone options are NUL-terminated but they use nd_printjnp (introduced in f6a0369d - that commit message even says that there is no NUL byte?).

I'm happy to fix up the newly-introduced calls.

Reply:
Regarding timezone options, Section 2 of RFC 4833 explicitly says:
> Note that they are NOT terminated by an ASCII NULL.

Commit f6a0369 perhaps could be refined through a separate follow-up.  Of course, there are many places for improvement in this source tree, some imperfections remain for a long time. That's a good reason not to introduce obvious imperfections with new code.